test/cpp/test_aten_xla_tensor.cpp

```diff
@@ -4691,19 +4691,17 @@ TEST_F(AtenXlaTensorTest, TestIndexAdd) {
               : torch::randint(100, value_sizes,
                                torch::TensorOptions(scalar_type));
       torch::Tensor result = torch::index_add(base, dim, index, value);
-      ForEachDevice(
-          {DeviceType::CPU, DeviceType::TPU}, [&](const torch::Device& device) {
-            torch::Tensor xla_base = CopyToDevice(base, device);
-            torch::Tensor xla_index = CopyToDevice(index, device);
-            torch::Tensor xla_value = CopyToDevice(value, device);
-            torch::Tensor xla_result =
-                torch::index_add(xla_base, dim, xla_index, xla_value);
-            AllEqual(result, xla_result);
+      ForEachDevice([&](const torch::Device& device) {
+        torch::Tensor xla_base = CopyToDevice(base, device);
+        torch::Tensor xla_index = CopyToDevice(index, device);
+        torch::Tensor xla_value = CopyToDevice(value, device);
+        torch::Tensor xla_result =
+            torch::index_add(xla_base, dim, xla_index, xla_value);
+        AllClose(result, xla_result);
 
-            ExpectCounterNotChanged("aten::.*", cpp_test::GetIgnoredCounters());
-            ExpectCounterChanged("xla::index_add_",
-                                 cpp_test::GetIgnoredCounters());
-          });
+        ExpectCounterNotChanged("aten::.*", cpp_test::GetIgnoredCounters());
```

Comment:
Move those before returning from function. Same other places you touched.

Reply:
Sorry I am a bit confuse. ExpectCounterNotChanged is still inside the `ForEachDevice` scope. The changes I made was replacing the `AllEqual` with `AllClose` and delete the `{DeviceType::CPU, DeviceType::TPU}` so GPU will also run this test. The rest was formatting tool adjusting the indent.

Reply:
These calls should be before the return.
You can leave them there and I will change them in another PR.

Reply:
Oh, I see your point. yea I will make the change